stack_overflow/src/main/java/com/gabrielcireap/stackOverflow/repository/data/DataUserRepository.java

```diff
@@ -0,0 +1,22 @@
+package com.gabrielcireap.stackOverflow.repository.data;
+
+import com.gabrielcireap.stackOverflow.entity.User;
+import com.gabrielcireap.stackOverflow.repository.UserRepository;
+import org.springframework.data.repository.Repository;
+
+import java.util.Optional;
+
+public interface DataUserRepository  extends Repository<User, Integer>, UserRepository {
+
+    void delete(User user);
+    @Override
+    default void remove(User user) {
+        delete(user);
+    }
+
+    Optional<User> findUserByUsernameAndPassword(String username, String password);
```

Comment:
Nice! Queries derived from method names + default methods for conforming to the old interface. Very good.